packages/plugin-electron-deliver-minidumps/minidump-loop.js

```diff
@@ -0,0 +1,86 @@
+const { readFile } = require('fs').promises
+
+module.exports = class MinidumpDeliveryLoop {
+  constructor (sendMinidump, onSend, minidumpQueue, logger) {
+    this._sendMinidump = sendMinidump
+    this._onSend = onSend
+    this._minidumpQueue = minidumpQueue
+    this._logger = logger
+    this._running = false
+  }
+
+  _onerror (err, minidump) {
+    this._logger.error('minidump failed to send…\n', (err && err.stack) ? err.stack : err)
+
+    if (err.isRetryable === false) {
+      this._minidumpQueue.remove(minidump)
+    }
+  }
+
+  async _readEvent (eventPath) {
+    // try to read the event associated with a minidump, or create an empty event if one doesn't exist
+    if (eventPath) {
+      try {
+        return JSON.parse(await readFile(eventPath, 'utf8'))
+      } catch (e) {
+        // swallow error - the minidump will be delivered without error info
+      }
+    }
+
+    return null
+  }
+
+  async _deliverMinidump (minidump) {
+    const event = await this._readEvent(minidump.eventPath)
+    const shouldSendMinidump = event && await this._onSend(event)
+
+    if (shouldSendMinidump === false) {
+      this._minidumpQueue.remove(minidump)
+      this._scheduleSelf()
+    } else {
+      try {
+        await this._sendMinidump(minidump.minidumpPath, event)
+      } catch (e) {
+        this._onerror(e, minidump)
```

Comment:
If `e.isRetryable === true` should we be retrying the delivery here? IIUC we're only ever attempting to deliver a minidump once